gsy_e_sdk/setups/Asset_API_scripts/rest_bid_offer_posting.py

```diff
@@ -0,0 +1,158 @@
+            ):
+                buy_energy = area_dict["asset_info"]["energy_to_buy"]
+                self.add_to_batch_commands.bid_energy_rate(
+                    asset_uuid=area_uuid, rate=10, energy=buy_energy
+                )
+
+                sell_energy = area_dict["asset_info"]["energy_to_sell"]
+                self.add_to_batch_commands.offer_energy_rate(
+                    asset_uuid=area_uuid, rate=10, energy=sell_energy
+                )
+
+            self.execute_batch_commands()
+
+    def on_event_or_response(self, message):
+        pass
+
+    def on_finish(self, finish_info):
+        self.is_finished = True
+
+
+def get_assets_name(indict: dict) -> dict:
+    """
+    Parse the grid tree and return all registered assets
+    wrapper for _get_assets_name
+    """
+    if indict == {}:
+        return {}
+    outdict = {"Area": [], "Load": [], "PV": [], "Storage": []}
+    _get_assets_name(indict, outdict)
+    return outdict
+
+
+def _get_assets_name(indict: dict, outdict: dict):
+    """
+    Parse the Collaboration / Canary Network registry
+    Return a list of the Market nodes the user is registered to
+    """
+    for key, value in indict.items():
+        if key == "name":
+            name = value
+        if key == "type":
+            area_type = value
+        if key == "registered" and value:
+            outdict[area_type].append(name)
+        if "children" in key:
+            for children in indict[key]:
+                _get_assets_name(children, outdict)
+
+
+aggr = Oracle(aggregator_name=ORACLE_NAME)
+AssetClient = RestAssetClient
+simulation_id = os.environ["API_CLIENT_SIMULATION_ID"]
+domain_name = os.environ["API_CLIENT_DOMAIN_NAME"]
+websockets_domain_name = os.environ["API_CLIENT_WEBSOCKET_DOMAIN_NAME"]
+asset_args = {"autoregister": False, "start_websocket": False}
+if AUTOMATIC:
+    registry = aggr.get_configuration_registry()
+    registered_assets = get_assets_name(registry)
+    load_names = registered_assets["Load"]
+    pv_names = registered_assets["PV"]
+    storage_names = registered_assets["Storage"]
+
+
+def register_asset_list(asset_names, asset_params, asset_uuid_map):
+    """Register the provided list of assets with the aggregator."""
+    for asset_name in asset_names:
+        print("Registered asset:", asset_name)
+        uuid = get_area_uuid_from_area_name_and_collaboration_id(
+            simulation_id, asset_name, domain_name
+        )
+        asset_params["asset_uuid"] = uuid
+        asset_uuid_map[uuid] = asset_name
+        asset = AssetClient(**asset_params)
+        asset.select_aggregator(aggr.aggregator_uuid)
+    return asset_uuid_map
+
+
+print()
+print("Registering assets ...")
+asset_uuid_mapping = {}
+asset_uuid_mapping = register_asset_list(load_names, asset_args, asset_uuid_mapping)
+asset_uuid_mapping = register_asset_list(pv_names, asset_args, asset_uuid_mapping)
+asset_uuid_mapping = register_asset_list(storage_names, asset_args, asset_uuid_mapping)
```

Comment:
It looks like here we're registering all assets (also PVs and storages), so this module is behaving differently than the redis one. I suppose we want to make them behave similarly, right?